### PR TITLE
fix: remove deprecated options

### PR DESCRIPTION
**Pull request by UziTech**
## Description

Remove deprecated options

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: docs/USING_ADVANCED.md [hunk 1]
```diff
@@ -54,16 +54,6 @@ console.log(marked.parse(markdownString));
...
-|sanitize (**deprecated**)|`boolean` |`false`  |v0.2.1   |If true, sanitize the HTML passed into `markdownString` with the `sanitizer` function.<br>**Warning**: This feature is deprecated and it should NOT be used as it cannot be considered secure.<br>Instead use a sanitize library, like [DOMPurify](https://github.com/cure53/DOMPurify) (recommended), [sanitize-html](https://github.com/apostrophecms/sanitize-html) or [insane](https://github.com/bevacqua/insane) on the output HTML! |
-|sanitizer  (**deprecated**)|`function`|`null`   |v0.3.4   |A function to sanitize the HTML passed into `markdownString`.|
-|smartypants (**deprecated**)|`boolean` |`false`  |v0.2.9   |Deprecated in v5.0.0 use [`marked-smartypants`](https://www.npmjs.com/package/marked-smartypants) to use "smart" typographic punctuation for things like quotes and dashes.|
-|xhtml (**deprecated**)|`boolean` |`false`  |v0.3.2   |Deprecated in v5.0.0 use [`marked-xhtml`](https://www.npmjs.com/package/marked-xhtml) to emit self-closing HTML tags for void elements (&lt;br/&gt;, &lt;img/&gt;, etc.) with a "/" as required by XHTML.|
```
Comment (styfle) [inline docs/USING_ADVANCED.md:66]: Can we keep this in the docs and change `deprecated in v5.0.0` to `removed in v6.0.0`?

Alternatively, we could remove here but add a new doc like `UPGRADING.md`.

Comment (UziTech) [reply docs/USING_ADVANCED.md:66]: I added a table below options with the extensions to migrate to.

**Review by styfle — APPROVED** [review]

File: docs/USING_ADVANCED.md (hunk 1 above)
Comment (calculuschild) [reply docs/USING_ADVANCED.md:66]: Can we include my "lite" smartypants extension? This just replicates Marked.js's built-in smartypants behavior without all the overhead of the full smartypants package.

https://www.npmjs.com/package/marked-smartypants-lite

Comment (UziTech) [reply docs/USING_ADVANCED.md:66]: Ya if you want to create a PR to add it as an option in that table I think that would be fine.

**Pull request merged**